source/iofs/iofs.go

```diff
@@ -0,0 +1,166 @@
+// +build go1.16
+
+package iofs
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path"
+	"strconv"
+
+	"github.com/golang-migrate/migrate/v4/source"
+)
+
+// Driver is a source driver that wraps io/fs#FS.
+type Driver struct {
+	migrations *source.Migrations
+	fsys       fs.FS
+	path       string
+}
+
+// NewDriver returns a new Driver from io/fs#FS and a relative path.
+func NewDriver(fsys fs.FS, path string) (source.Driver, error) {
```

Comment:
Rename to `New` to follow the same pattern as [httpfs.New()](https://pkg.go.dev/github.com/golang-migrate/migrate/v4@v4.13.0/source/httpfs#New)